tests/UnitTest.php

```diff
@@ -58,41 +65,85 @@ public function testCreateMediaWithDefaultPreviewDrivers()
         );
 
         $this->assertEqualsFixture('create_media_with_set_preview_drivers.json', $media->toArray());
+
+        Storage::assertExists('preview_file.png');
+
+        self::$mediaTestState->assertChangesEqualsFixture('create_media_with_default_preview_drivers.json');
     }
 
     public function testCreateBulkMediaWithSetPreviewDrivers()
     {
-        $this->mockGenerateFilename();
+        $this->mockGenerateFilename(
+            [
+                'argument' => 'file.png',
+                'result' => 'file1.png',
+            ],
+            [
+                'argument' => 'file.png',
+                'result' => 'file2.png',
+            ],
+        );
 
         $mediaArray = [
             'media' => [
-                'file' => self::$file,
+                [
+                    'file' => self::$file,
+                ],
+                [
+                    'file' => self::$file,
+                ]
             ],
         ];
 
         $media = app(MediaServiceContract::class)->bulkCreate(
-            data: $mediaArray,
+            data: $mediaArray['media'],
             previewDrivers: PreviewDriverEnum::File,
         );
 
-        $this->assertEqualsFixture('bulk_create_media_with_set_preview_drivers.json', $media['media']->toArray());
+        $result = array_map(fn($item) => $item->toArray(), $media);
+
+        Storage::assertExists(['preview_file1.png', 'preview_file2.png']);
+
+        $this->assertEqualsFixture('bulk_create_media_with_set_preview_drivers.json', $result);
+
+        self::$mediaTestState->assertChangesEqualsFixture('create_bulk_media_with_set_preview_drivers.json');
     }
 
     public function testCreateBulkMediaWithDefaultPreviewDrivers()
     {
-        $this->mockGenerateFilename();
+        $this->mockGenerateFilename(
+            [
+                'argument' => 'file.png',
+                'result' => 'file1.png',
+            ],
+            [
+                'argument' => 'file.png',
+                'result' => 'file2.png',
+            ],
+        );
 
         $mediaArray = [
             'media' => [
-                'file' => self::$file,
+                [
+                    'file' => self::$file,
+                ],
+                [
+                    'file' => self::$file,
+                ]
             ],
         ];
 
         $media = app(MediaServiceContract::class)->bulkCreate(
-            data: $mediaArray,
+            data: $mediaArray['media'],
             previewDrivers: PreviewDriverEnum::File,
         );
 
-        $this->assertEqualsFixture('bulk_create_media_with_set_preview_drivers.json', $media['media']->toArray());
+        $result = array_map(fn($item) => $item->toArray(), $media);
+
+        Storage::assertExists(['preview_file1.png', 'preview_file2.png']);
+
+        $this->assertEqualsFixture('bulk_create_media_with_set_preview_drivers.json', $result);
```

Comment:
could we reduce array_map by this?

```suggestion
        $this->assertEqualsFixture('bulk_create_media_with_set_preview_drivers.json', $result->toArray());
```

Reply:
@DenTray 
$result variable is array